File: BackendRecycle/routes/empleadosRoutes.js

const EmpleadosController = require('../controllers/empleadosController')

const passport = require('passport')

module.exports = (app, upload) => {

    app.get('/api/recycle/empleados/getAll/:is_available', passport.authenticate('jwt', {session: false}), EmpleadosController.getAll)
    //GET => Traer Datos
    app.get('/api/recycle/empleados/findByRol/:id_rol', passport.authenticate('jwt', {session: false}), EmpleadosController.findByRol)
    
    //app.get('/api/recycle/empleados/getAll', EmpleadosController.getAll)

    //POST => Guardar Datos
    app.post('/api/recycle/empleados/create', EmpleadosController.register)
    app.post('/api/recycle/empleados/addUser/:id_rol', passport.authenticate('jwt', {session: false}), upload.array('image', 1), EmpleadosController.addUser)
    app.post('/api/recycle/empleados/addUserWithoutImage/:id_rol', passport.authenticate('jwt', {session: false}), EmpleadosController.addUserWithoutImage)
      

    app.post('/api/recycle/empleados/login', EmpleadosController.login)

    //PUT => ACTUALIZA DATOS
    //401 UNAUTHORIZED
    app.put('/api/recycle/empleados/update', passport.authenticate('jwt', {session: false}), upload.array('image', 1), EmpleadosController.update)
    app.put('/api/recycle/empleados/desactivarEmpleado', passport.authenticate('jwt', {session: false}), EmpleadosController.desactivarEmpleado)
    app.put('/api/recycle/empleados/reactivarEmpleado', passport.authenticate('jwt', {session: false}), EmpleadosController.reactivarEmpleado)
    
    //app.put('/api/recycle/empleados/update', upload.array('image', 1), EmpleadosController.update)

    app.put('/api/recycle/empleados/updateWithoutImage',passport.authenticate('jwt', {session: false}), EmpleadosController.updateWithoutImage)
}

File: RecycleApp/app/src/main/java/com/seminariomanufacturadigital/recycleapp/routers/EmpleadosRoutes.kt
package com.seminariomanufacturadigital.recycleapp.routers

import com.seminariomanufacturadigital.recycleapp.models.Empleado
import com.seminariomanufacturadigital.recycleapp.models.ResponseHttp
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Part
import retrofit2.http.Path

//Clase(Interface) 2 para conectar el backend con la app
interface EmpleadosRoutes {
    @POST("empleados/create") //La url que se  necesita para crear un nuevo empleado
    fun  register(@Body empleado: Empleado
    ): Call<ResponseHttp>

    @FormUrlEncoded
    @POST("empleados/login")
    fun login(@Field("no_empleado") numEmpleadc: String,
              @Field("contrasena") contrasena:String
    ) :Call<ResponseHttp>

    @Multipart
    @PUT("empleados/update")
    fun   update(
        @Part image: MultipartBody.Part,
        @Part("empleado") empleado: RequestBody,
        @Header("Authorization") token: String
    ): Call<ResponseHttp>

    @PUT("empleados/updateWithoutImage")
    fun updateWithoutImage(
        @Body empleado: Empleado,
        @Header("Authorization") token: String
    ): Call<ResponseHttp>

    @GET("empleados/findByRol/{id_rol}")
    fun findByRol(
        @Path("id_rol") idRol: String,
        @Header("Authorization") token: String
    ): Call<ArrayList<Empleado>>

    @Multipart
    @POST("empleados/addUser/{id_rol}") //La url que se  necesita para crear un nuevo empleado
    fun  addUser(
        @Path("id_rol") idRol: String,
        @Part imagen: MultipartBody.Part,
        @Part("empleado") empleado: RequestBody,
        @Header("Authorization") token: String
    ): Call<ResponseHttp>

    @POST("empleados/addUserWithoutImage/{id_rol}") //La url que se  necesita para crear un nuevo empleado
    fun addUserWithoutImage(
        @Path("id_rol") idRol: String,
        @Body empleado: Empleado,
        @Header("Authorization") token: String
    ): Call<ResponseHttp>
    @GET("empleados/getAll/{is_available}")
    fun getAll(
        @Path("is_available") is_available:Boolean,
        @Header("Authorization") token: String
    ): Call<ArrayList<Empleado>>

    @PUT("empleados/desactivarEmpleado")
    fun desactivarEmpleado(
        @Body empleado: Empleado,
        @Header("Authorization") token: String
    ): Call<ResponseHttp>

    @PUT("empleados/reactivarEmpleado")
    fun reactivarEmpleado(
        @Body empleado: Empleado,
        @Header("Authorization") token: String
    ): Call<ResponseHttp>

}